controller/appcontroller.go

```diff
@@ -913,20 +946,34 @@ func (ctrl *ApplicationController) autoSync(app *appv1.Application, syncStatus *
 
 // alreadyAttemptedSync returns whether or not the most recent sync was performed against the
 // commitSHA and with the same app source config which are currently set in the app
-func alreadyAttemptedSync(app *appv1.Application, commitSHA string) bool {
+func alreadyAttemptedSync(app *appv1.Application, commitSHA string) (bool, appv1.OperationPhase) {
```

Comment:
why do you need the extra return? this value must be in scope for any caller

Reply:
I wanted to have `app.Status.OperationState != nil`  as close as possible `app.Status.OperationState.Phase` and don't assume that `app.Status.OperationState` cannot be nil if `alreadyAttemptedSync` returns true